src/transformers/hf_api.py

```diff
@@ -146,15 +151,20 @@ def presign(self, token: str, filename: str, organization: Optional[str] = None)
         d = r.json()
         return PresignedUrl(**d)
 
-    def presign_and_upload(self, token: str, filename: str, filepath: str, organization: Optional[str] = None) -> str:
+    def presign_and_upload(
+        self, token: str, filetype: str, filename: str, filepath: str, organization: Optional[str] = None
+    ) -> str:
         """
         HuggingFace S3-based system, used for datasets and metrics.
 
         Get a presigned url, then upload file to S3.
 
         Outputs: url: Read-only url for the stored file on S3.
         """
-        urls = self.presign(token, filename=filename, organization=organization)
+        assert filetype in self.ALLOWED_S3_FILE_TYPES, "Please specify filetype from {}".format(
```

Comment:
(nit) capital letter for "ALLOWED_S3_FILE_TYPES" for class attribute is a bit weird for me. I'd prefer `allowed_s3_file_types`

Reply:
backported from code in `datasets` so staying consistent